up_client_mqtt5_python/mqtt5_utransport.py

```diff
@@ -0,0 +1,396 @@
+                        Base64ProtobufSerializer().deserialize(
+                            uattributes_proto.id.SerializeToString()
+                        ),
+                    )
+                )
+            publish_properties.UserProperty.append(
+                ("2", UMessageType.Name(uattributes_proto.type))
+            )
+            if uattributes_proto.HasField("source"):
+                publish_properties.UserProperty.append(
+                    (
+                        "3",
+                        ShortUriSerializer().serialize(
+                            uattributes_proto.source
+                        ),
+                    )
+                )
+            if uattributes_proto.HasField("sink"):
+                publish_properties.UserProperty.append(
+                    (
+                        "4",
+                        ShortUriSerializer().serialize(uattributes_proto.sink),
+                    )
+                )
+            publish_properties.UserProperty.append(
+                ("5", UPriority.Name(uattributes_proto.priority))
+            )
+            if uattributes_proto.HasField("ttl"):
+                publish_properties.UserProperty.append(
+                    ("6", str(uattributes_proto.ttl))
+                )
+            if uattributes_proto.HasField("permission_level"):
+                publish_properties.UserProperty.append(
+                    ("7", str(uattributes_proto.permission_level))
+                )
+            if uattributes_proto.HasField("commstatus"):
+                publish_properties.UserProperty.append(
+                    ("8", UCode.Name(uattributes_proto.commstatus))
+                )
+            if uattributes_proto.type == UMessageType.UMESSAGE_TYPE_RESPONSE:
+                publish_properties.UserProperty.append(
+                    (
+                        "9",
+                        Base64ProtobufSerializer().deserialize(
+                            uattributes_proto.reqid.SerializeToString()
+                        ),
+                    )
+                )
+            if uattributes_proto.HasField("token"):
+                publish_properties.UserProperty.append(
+                    ("10", uattributes_proto.token)
+                )
+            if uattributes_proto.HasField("traceparent"):
+                publish_properties.UserProperty.append(
+                    ("11", uattributes_proto.traceparent)
+                )
+        except ValueError as e:
+            raise ValueError(e) from e
+
+        payload: bytes = payload_proto.SerializeToString()
+        self._mqtt_client.publish(
+            topic=self.mqtt_topic_builder(
+                topic=message.attributes.source
+            ),
+            payload=payload,
+            qos=1,
+            properties=publish_properties,
+        )
+
+        return UStatus(code=UCode.OK, message="OK")
+
+    def register_listener(self, topic: UUri, listener: UListener) -> UStatus:
```

Comment:
I believe that there is a PR currently out for 1.5.8 that updates the register_listener, unregister_listener and receive functions to use a source_filter and a sink_filter. @sophokles73 and @stevenhartley, is this correct that the uTransport spec change is expected to get in for 1.5.8?

Reply:
yes, the corresponding PR is https://github.com/eclipse-uprotocol/up-spec/pull/134 which will go into 1.5.8